### PR TITLE
RUBY-2147 Run FLE tests on replica set

### PR DESCRIPTION
**Pull request by egiurleo**
Changes in this PR:
- Run FLE tests on a replica set in CI
- In the FLE prose tests, perform all key vault operations using write concern majority
- Fix all instances of db/collection names as symbols.

### CODE REVIEW

File: lib/mongo/crypt/encryption_io.rb [hunk 1]
```diff
@@ -179,7 +179,8 @@ def validate_key_vault_namespace!(key_vault_namespace)
       def key_vault_collection
         @key_vault_collection ||= @key_vault_client.with(
           database: @key_vault_db_name,
-          read_concern: { level: :majority }
+          read_concern: { level: :majority },
+          write_concern: { w: :majority }
```
Comment (egiurleo) [inline lib/mongo/crypt/encryption_io.rb]: At some point during refactoring, I had removed this line.

File: spec/support/crypt.rb [hunk 2]
```diff
@@ -52,6 +52,13 @@ module Crypt
 
     # Example value to encrypt
     let(:ssn) { '123-456-7890' }
+
+    let(:key_vault_collection) do
+      authorized_client.with(
+        database: key_vault_db,
+        write_concern: { w: :majority }
+      )[key_vault_coll]
+    end
```
Comment (egiurleo) [inline spec/support/crypt.rb]: From the [prose tests](https://github.com/mongodb/specifications/tree/master/source/client-side-encryption/tests): Perform all applicable operations on key vault collections (e.g. inserting an example data key, or running a find command) with readConcern/writeConcern "majority".

**Comment by egiurleo** [comment]
This definitely makes the test suite take a bit longer. I'd be happy to talk about some ways to cut down on FLE testing if it would reduce testing time.

File: spec/integration/client_side_encryption/auto_encryption_reconnect_spec.rb [hunk 3]
```diff
@@ -200,7 +200,10 @@
 
   context 'with custom key vault client option' do
     let(:key_vault_client_option) do
-      Mongo::Client.new(SpecConfig.instance.addresses).use(:test)
+      new_local_client(
```
Comment (p-mongo) [inline spec/integration/client_side_encryption/auto_encryption_reconnect_spec.rb]: :+1:

File: spec/mongo/client_encryption_spec.rb [hunk 4]
```diff
@@ -1,14 +1,13 @@
...
 
 describe Mongo::ClientEncryption do
   require_libmongocrypt
   include_context 'define shared FLE helpers'
 
   let(:client) do
     ClientRegistry.instance.new_local_client(
-      [SpecConfig.instance.addresses.first]
+      SpecConfig.instance.addresses,
+      SpecConfig.instance.test_options
```
Comment (p-mongo) [inline spec/mongo/client_encryption_spec.rb]: :+1:

**Pull request merged**